peps/pep-9999.rst

```diff
@@ -0,0 +1,233 @@
+
+
+``f(a, b, *, x)``
+'''''''''''''''''''''''''''''''''''''''''''''
+On a few occasions the idea has been floated to borrow the syntax from keyword-only function definitions. This is less arbitrary than ``f(%x)`` or variants, but no less so than ``f(x=)``. 
+
+However, we object that:
+
+* For any given argument, it is less clear from local context whether it is positional or named. The ``*`` could easily be missed in a long argument list and named arguments may be read as positional or vice versa.
+* It is unclear whether keyword arguments for which the value was not elided may follow the ``*``. If so, then their relative position will be inconsistent but if not, then an arbitrary grouping is enforced between different types of keyword arguments.
+
+
+Objections
+==================
+
+There are only a few hard objections to the introduction of this syntactic sugar. Most of those not in favour of this feature are simply in the camp of 'I wouldn't use it'. However over the extensive conversations about this feature, the following objections were the most common:
+
+The syntax is ugly
+--------------------------------------------
+This objection is by far the most common. On the contrary, we argue that:
+
+* This objection is is subjective and many community members disagree
+* This is already established syntax used in f-strings
+* Programmers will, as ever, adjust over time
+
+The feature is confusing
+--------------------------------------------
+We argue that:
+
+* Introducing new features typically has this temporary impact
+* The syntax is very similar to the established ``f'{x=}'`` syntax
+* The feature and syntax are familiar from other popular modern languages
+*  The expansion of ``x=`` to ``x=x`` is in fact a trivial feature and inherently significantly less complex than ``*arg`` and ``**kwarg`` expansion
+* This particular syntactic form has been independently proposed on numerous occasions, indicating that it is the most obvious [8]_ [9]_ [13]_
+
+The feature is not explicit
+--------------------------------------------
+This is based on a misunderstanding of the Zen of Python. Keyword arguments are fundamentally more explicit than positional ones where argument assignment is only visible at the function definition. On the contrary, the proposed syntactic sugar contains all the information as is conveyed by the established keyword argument syntax but without the redundancy. Moreover, the introduction of this syntactic sugar incentivises use of keyword arguments, making practical Python code more explicit.
+
+The feature adds another way of doing things
+--------------------------------------------
+
+The same argument can be made against all syntax changes. This is a simple syntactic sugar, much as ``x += 1`` is sugar for ``x = x + 1``. This isn't tantamount to a 'new way' of passing arguments but a more readable notation for the same way.
```

Comment:
Notably, `x += 1` is not mere syntactic sugar for `x = x + 1`; this can be demonstrated with lists, where `x = [1,2]; x += [3,4]` will mutate the list in place, but `x = x + [3,4]` will not. Here's an example where it really truly is syntactic sugar: https://peps.python.org/pep-3135/ Although this particular one, while quite popular, has raised some eyebrows for being too magical and not something to be replicated.

Reply:
I think the reason this "feels wrong" is that adding another way of doing things isn't the problem. Python's Zen does not say "there's only one way to do things, you do it our way or you don't do it at all". The Zen says that there is one OBVIOUS way to do things. Given that Python was Turing-complete before this proposal (or any other), by definition ANY proposal is adding another way of doing things. Maybe this entire paragraph is arguing against something that doesn't matter?

Perhaps, if you want to argue about syntactic sugar's benefit, the matrix multiplication operator would be a good place to start. Numpy users for a long time were able to do both types of multiplication, but one of them had an operator and the other was done with a method. Could you do what you wanted? Sure. Was it the way you wanted to write your code? No. And when code isn't as expressive as you'd like, there's the possibility of bugs slipping in.

The benefit of code being obvious is precisely that, in fact: bugs become visible. It is trivially easy to see when you have simple code that's doing a simple task, and you can equally easily see if it's failing to do that task. This proposal achieves that by raising from the concrete "take variable spamination and pass it as the named parameter spamination" to the abstract "pass named parameter spamination as itself".